src/nvme/mi.c

```diff
@@ -41,18 +41,26 @@ static bool nvme_mi_probe_enabled_default(void)
  */
 nvme_root_t nvme_mi_create_root(FILE *fp, int log_level)
 {
-	struct nvme_root *r = calloc(1, sizeof(*r));
+	struct nvme_root *r;
 
+	r = calloc(1, sizeof(*r));
 	if (!r) {
+		errno = ENOMEM;
 		return NULL;
 	}
-	r->log_level = log_level;
-	r->fp = stderr;
+
+	if (!fp)
+		fp = stderr;
```

Comment:
Seems usually the input argument `fp` not used as a local variable and also `r->log.fp` can be set `stderr` directly as below.
```c
    r->log.fp = fp ? fp : stderr;
```